jvm-packages/xgboost4j-spark/src/main/scala/org/apache/spark/SparkParallelismTracker.scala

```diff
@@ -98,31 +98,30 @@ private[spark] class TaskFailedListener extends SparkListener {
     taskEnd.reason match {
       case taskEndReason: TaskFailedReason =>
         logger.error(s"Training Task Failed during XGBoost Training: " +
-            s"$taskEndReason, stopping SparkContext")
-        TaskFailedListener.startedSparkContextKiller()
+            s"$taskEndReason, cancelling all jobs")
+        TaskFailedListener.cancelAllJobs()
```

Comment:
1. I don't think we should change the default behavior of killing sparkcontext, over the year, a lot of fault recovery strategy in prod env have been built based on killing context behavior...instead, we should provide both behavior. Even more, cancel the job is just for a small percentage of customers who train multiple XGB models simultaneously or train various models in the same app.

2. here you called cancelAllJobs, I don't think there is any difference than killing SparkContext, you should call 
![image](https://user-images.githubusercontent.com/678008/90417741-021f1200-e069-11ea-95a1-a4f12f0c2f44.png)

Reply:
Thx @CodingCat. for
> I don't think we should change the default behavior of killing sparkcontext, over the year, a lot of fault recovery strategy in prod env have been built based on killing context behavior...instead, we should provide both behavior. Even more, cancel the job is just for a small percentage of customers who train multiple XGB models simultaneously or train various models in the same app.

Yeah, make sense. I will update the change to add a parameter to control for both scenarios.

> here you called cancelAllJobs, I don't think there is any difference than killing SparkContext, you should call

The intent of this PR is to throw an Exception when doing cancelAllJobs. User should try catch
``` scala
try {
val model = xgboostClassifier.fit(dataset)
} catch {
case e: Exception =>
}
```

1. **`cancelAllJobs`**
``` scala
  private[scheduler] def doCancelAllJobs(): Unit = {
    // Cancel all running jobs.
    runningStages.map(_.firstJobId).foreach(handleJobCancellation(_,
      Option("as part of cancellation of all jobs")))
    activeJobs.clear() // These should already be empty by this point,
    jobIdToActiveJob.clear() // but just in case we lost track of some jobs...
  }
```
2. **`cancelJob`**
``` scala
  private[scheduler] def handleJobCancellation(jobId: Int, reason: Option[String]): Unit = {
    if (!jobIdToStageIds.contains(jobId)) {
      logDebug("Trying to cancel unregistered job " + jobId)
    } else {
      failJobAndIndependentStages(
        jobIdToActiveJob(jobId), "Job %d cancelled %s".format(jobId, reason.getOrElse("")))
    }
  }
``` 
From the code, `cancelAllJobs` will finally call the handleJobCancellation just like `cancelJob` does.